### PR TITLE
When decompressing, count the elements

### PR DESCRIPTION
**Pull request by graeme-winter**
If we provide input which should be the compressed equivalent of `slow*fast` values, verify this.

### CODE REVIEW

**Comment by ndevenish** [comment]
Looks reasonable - although I wonder if this is a good time to clear up the buffer overflow vulnerability? e.g. pass in a "maximum" size and get back the actual number read.

**Comment by graeme-winter** [comment]
> Looks reasonable - although I wonder if this is a good time to clear up the buffer overflow vulnerability? e.g. pass in a "maximum" size and get back the actual number read.

Wilco

File: boost_python/compression.cc [hunk 1]
```diff
@@ -117,6 +118,9 @@ unsigned int dxtbx::boost_python::cbf_decompress(const char *packed,
   bool le = little_endian();
 
   while (j < packed_sz) {
+    if ((values - original) > unpacked_sz) {
+      break;
+    }
```
Comment (Anthchirp) [inline boost_python/compression.cc:123]: should this `return 0`?

Comment (graeme-winter) [reply boost_python/compression.cc:123]: 🤔

TBH I think this should probably just raise an exception here, but I am not sure

It probably _should not_ return 0 as it _has_ unpacked a whole image - what we should do with trailing garbage is an interesting question

Comment (ndevenish) [reply boost_python/compression.cc]: I think "number read" is fine - in theory you might want to know
- if you got less than expected
- If you filled a buffer if for some reason you don't know how long
Maybe there's a better signal e.g. returning one larger but that sounds like it'd just cause problems when forgetting. This probably isn't used enough for it to be a problem.

**Pull request merged**